update/vm.sls

```diff
@@ -0,0 +1,55 @@
+{% if grains['os'] == 'Debian' %}
+# mitigation for DSA-4371-1 (CVE-2019-3462)
+/etc/apt/sources-dsa-4371-1.list:
+  file.managed:
+    - contents:
+      - deb http://cdn-fastly.deb.debian.org/debian-security {{ grains['oscodename'] }}/updates main
+
+/etc/apt/apt.conf.d/99dsa-4371-1:
+  file.managed:
+    - contents: |
+        Dir::Etc::SourceList "/etc/apt/sources-dsa-4371-1.list";
+        Dir::Etc::SourceParts "";
+        Acquire::http::AllowRedirect "false";
+    - require:
+      - file: /etc/apt/sources-dsa-4371-1.list
+
+{%- if grains['oscodename'] == 'stretch' %}
+{%- set pkg = "libapt-pkg5.0" %}
+{%- set pkgver = "1.4.9" %}
+{%- elif grains['oscodename'] == 'jessie' %}
+{%- set pkg = "libapt-pkg4.12" %}
+{%- set pkgver = "1.0.9.8.5" %}
+{%- else %}
+# sid, buster, unknown
+{%- set pkg = "libapt-pkg5.0" %}
+{%- set pkgver = "1.8.0~alpha3.1" %}
+{%- endif %}
+
+{{pkg}}:
+  pkg.installed:
+    - version: {{pkgver}}
+    - refresh: True
+    - allow_updates: True
+    - require:
+      - file: /etc/apt/apt.conf.d/99dsa-4371-1
+
+# remove the config after, but only if update was successful
+# avoid file.missing as it would have the same name as the state creating the
+# file
+rm -f /etc/apt/apt.conf.d/99dsa-4371-1:
+  cmd.run:
+    - require:
+      - pkg: {{pkg}}
+{% endif %}
+
+
+update:
+  pkg.uptodate:
+    - refresh: True
```

Comment:
In my bash script I do a update run with an empty source.list first to ensure that all old list files get cleaned up. AFAIK, on network errors `apt-get update` keeps the old list files. OTOH probably nobody has `cdn-fastly.deb.debian.org` in their source list (the probably use `deb.debian.org` instead), so there should be no old lists available.

Reply:
Does it matter? We check for explicit package version, not "the latest". So, on network error, the state will fail to find the right version (I've checked by putting 1.4.10 there).

Reply:
Yes, AFAIU this bug can also be used on other downloaded files. So you can replace the `Packages` list with another version containing another hash for the "right" version.

Reply:
`apt clean` call before?

Reply:
`apt clean` does not remove the package lists (you normally need them). AFAIK you need to run apt-get update with empty sources.list or manually delete stuff in `/var/lib/apt/lists`.